src/status_im/ui/screens/routing/core.cljs

```diff
@@ -117,27 +119,45 @@
-                                    (let [route      (aget routes idx)
-                                          route-name (keyword (.-routeName route))]
-                                      (tabbar/minimize-bar route-name)))))}
-          (prepare-config config))))
+  (let [res (nav-reagent/stack-navigator
+             routes
+             (merge {:headerMode        "none"
+                     :cardStyle         {:backgroundColor :white}
+                     #_:transitionConfig
+                     #_(fn []
+                         #js {:transitionSpec #js{:duration 10}})
+                     :onTransitionStart (fn [n]
+                                          (let [idx    (.. n
+                                                           -navigation
+                                                           -state
+                                                           -index)
+                                                routes (.. n
+                                                           -navigation
+                                                           -state
+                                                           -routes)]
+                                            (when (and (array? routes) (int? idx))
+                                              (let [route      (aget routes idx)
+                                                    route-name (keyword (.-routeName route))]
+                                                (tabbar/minimize-bar route-name)))))}
+                    (prepare-config config)))
+        default-get-state-for-action  (.-getStateForAction (.-router res))
+        new-get-state-for-action (fn [action state]
+                                   ;; Override default getStateForAction on this stack navigator
+                                   ;; if we have a custom event set in wizard-back-event atom
+                                   (if (and (= (.-type action) (.-BACK navigation/navigation-actions))
+                                            @screens.navigation/wizard-back-event)
+                                     (if @screens.navigation/processing-back-event?
+                                       (do
+                                         (reset! screens.navigation/processing-back-event? false)
+                                         (default-get-state-for-action action state))
+                                       (do
+                                         (reset! screens.navigation/processing-back-event? true)
+                                         (re-frame/dispatch @screens.navigation/wizard-back-event)
+                                         ;; Return nil so that BACK event processing ends here
+                                         nil))
+                                     (default-get-state-for-action action state)))]
+    (set! (-> res .-router .-getStateForAction) new-get-state-for-action)
```

Comment:
can't you do a conditional set instead of doing the conditional in the new-get-state-for-action?